### DISCUSSION TITLE
Elsa performance benchmark report

### DISCUSSION THREAD

**Discussion by neerajmathur** [elsa-workflows/elsa-core#4420] [Q&A] [answered]
HI,

We are planning to use a low code platform to create integration workflows between API's. And our performance requirement are little stringent. Execution of a workflow should not take more than 220ms.

Use case we are planning
1. Get an HTTP request
2. Perform various validation check of input data, For example it is a valid request from authentic source, This mostly involves, JSON parsing and transformation steps, few Javascript steps and call API endpoint to get required data.
3. Transform it back to source format.
4. Write HTTP response

We are planning to use ELSA in production for this purpose.

Questions:
1. Is there a performance bench mark report published?
2. How reliable it is for production use when compared to Apache Camel?

We really liked ELSA framework, However when come to production use we need reliability artifacts to show against the proven system like Apache Camel.

**Comment by sfmskywalker** [comment, marked as answer]
Hi @neerajmathur 

1. I'm not aware of any performance benchmarks that are publicly available. However, if you need that kind of performance, you might want to consider doing a POC with [Elsa 3](https://v3.elsaworkflows.io/) in addition to Elsa 2.
2. I'm not familiar with Apache Camel, but from personal experience Elsa 2 is very reliable. Elsa 3 hasn't hit real production environments yet, so its reliability remains to be proven still. It's currently in a prerelease stage, with 3.0 expected to be GA towards the end of this year, possibly around January.

I wish I could have offered concrete reliability artefacts, but I don't have them yet. In any case, good luck with the project!

**Comment by neerajmathur** [comment]
Thank You for an update.  Apache Camel (https://camel.apache.org/ ) is a popular open source integration framework based on Enterprise Integration Patterns. 

I am following ELSA from last 1 year and impressed by it's progress and roadmap directions. Hence looking to replace our home grown basic engine with ELSA Core. Hence, looking for all these proofs to support my argument.
